Azure/azure-cli
# [APIM] `az apim api create`: Parse URL to detect Protocol and API type

**Related command**


```
az apim api create --resource-group {} --service-name {} --api-id {} --display-name {} --path {} --service-url {}
```

**Description**


The subcommand to create a new API in APIM (`az apim api create`) defaults the protocols and api type values to `https` and `http` by default, making the respective flags to be optional. But with support added for websockets, there is a new set of protocols (`ws` / `wss`) and api type (`websocket`) required when adding a new websocket API.

Since both of these can be derived from the service url parameter passed, this PR adds the ability to infer the protocol and api type while still defaulting to the previous behavior if unknown.

resolves #24717

**Testing Guide**


```
az apim api create --resource-group {} --service-name {} --api-id {} --display-name {} --path {} --service-url wss://httpbin.org
```

This command will fail in the current version since the default values for protocols and api type would be invalid as called out in [this issue](https://github.com/Azure/azure-cli/issues/24717).

With the changes from this PR, both would be inferred from the service url (the scheme specifically; `wss`).

---

This checklist is used to make sure that common guidelines for a pull request are followed.
